ui/plugin-lifecycle/src/main/services/plugin-publisher.service.ts

```diff
@@ -0,0 +1,179 @@
+import { Injectable } from "@angular/core";
+import { Observable } from "rxjs";
+import { ChangeScopePlugin } from "../interfaces/Plugin";
+import { ChangeScopeRequest } from "../classes/ChangeScopeRequest";
+import { ChangeTenantScopeService } from "./change-tenant-scope.service";
+import { ScopeFeedback } from "../classes/ScopeFeedback";
+import { ChangeScopeItem } from "../interfaces/ChangeScopeItem";
+import { UiPluginMetadataResponse } from "@vcd/bindings/vcloud/rest/openapi/model";
+import { HttpResponse } from "@angular/common/http";
+import { PluginService } from "./plugin.service";
+import { UiPluginTenantsResponse } from "../interfaces/Tenant";
+
+@Injectable()
+export class PluginPublisher {
+    constructor(
+        private changeOrgScopeService: ChangeTenantScopeService,
+        private pluginService: PluginService
+    ) {}
+
+    /**
+     * Toggle the publish state of a plugin for all organisations.
+     * @param plugins list of plugins
+     * @param url the base url where the request will be made
+     * @param hasToBe string value which says "published" or "unpublished"
+     * @param trackScopeChange determinate will this request be tracked
+     */
+    private generatePublishRequestForAllTenants(
+        plugins: UiPluginMetadataResponse[], url: string, hasToBe: string, trackScopeChange: boolean = false
+    ): Observable<HttpResponse<UiPluginTenantsResponse[]>> {
+        // Register change scope list
+        let changeScopeRequests: ChangeScopeRequest[];
+
+        // If the requests are not trackable create list for them
+        if (!trackScopeChange) {
+            changeScopeRequests = [];
+        }
+
+        // Loop throught the plugins list
+        plugins.forEach((pluginToUpdate) => {
+            // Create request url
+            const REQ_URL = `${url}/cloudapi/extensions/ui/${pluginToUpdate.id}/tenants/${hasToBe}`;
```

Comment:
why does the absolute URL need to be the identification for the tracker?  `pluginToUpdate.id` uniquely identifies a plugin.

Reply:
There is no plugin id in the response object.